models/ims-model.js

```diff
@@ -0,0 +1,1692 @@
+    DocumentType : 1,
+    DocumentName : 'test.txt'
+  }
+};
+
+let notes = {
+  FWTNoteToParentRef : {
+    ParentId : null,
+    ParentType : 0,
+    NoteDetails : {
+      Text: null,
+        NoteLabels : {
+          NoteLabel : 'test note label'
+        },
+       NoteAttachments : {
+         NoteAttachmentList : {
+          AttachmentName : 'test.txt',
+          AttachmentIdentifier : null,
+          AttachmentType: 0
+         }
+       }
+    }
+  }
+};
+
+const createClient = () =>
+  new Promise((resolve, reject) =>
+    soap.createClient(wsdlUrl, { wsdl_headers: { Authorization: auth } }, (err, client) => {
+      if (err)
+        return reject(err);
+      client.setEndpoint(config.ims.endpoint);
+      client.setSecurity(new soap.BasicAuthSecurity(config.ims.apiUser, config.ims.apiPassword));
+      return resolve(client);
+    }
+  )
+);
+
+/************************************************************
+ * @param {Object} caseType type of case to create
+ * @param {Object} caseType.FWTCaseCreate
+ * @param {String} caseType.FWTCaseCreate.ClassificationEventCode
+ ************************************************************/
+const createCase = (client, caseType) =>
+  new Promise((resolve, reject) =>
+      client.createCase(caseType,
+      (err, result) => {
+          if (err)
+          return reject(err);
+      console.log('Case reference: ' + JSON.stringify(result, null, 2));
+      return resolve(result);
+    }
+  )
+);
+
+/************************************************************
+ * @param {Object} eForm Case eForm to add
+ * @param {Object} eForm.FWTCaseEformNew
+ * @param {Object} eForm.FWTCaseEformNew.CaseReference
+ * @param {String} eForm.FWTCaseEformNew.EformName
+ ************************************************************/
+  const addCaseForm = (client, caseRef) =>
+    new Promise(function(resolve, reject){
+      eForm.FWTCaseEformNew.CaseReference =
+      eForm.FLCaseEformInstance.CaseReference = caseRef;
+      client.addCaseEform(eForm,
+        (err, result) => {
+          if (err)
+            return reject(err);
+
+          return resolve(result);
+        }
+      )
+    });
+
+  /************************************************************
+   * @param {Object} eformData Case eForm data
+   * @param {Object} eformData.FLEformFields
+   * @param {Object} eformData.FLEformFields.CaseEformInstance
+   * @param {String} eformData.FLEformFields.CaseEformInstance.CaseReference
+   * @param {String} eformData.FLEformFields.CaseEformInstance.EformName
+   * @param {String} eformData.FLEformFields.EformData
+   ************************************************************/
+  const writeFormData = (client, caseRef) =>
+    new Promise(function(resolve, reject) {
+      let ids = {};
+      eformData.FLEformFields.EformData.EformFields.map((item, index) => {ids[item.FieldName] = index})
+      eformData.FLEformFields.CaseEformInstance.CaseReference =
```

Comment:
Could you explain this bit?

Reply:
I have now refactored this code to add the fields that are required. In this version the fields map is indexed and the Case Reference in the eForm is being set. (At this point I wasn't sure which case reference, num to set so I set them all) :)